src/examples/java/software/amazon/encryption/s3/examples/ClientConfigurationExample.java

```diff
@@ -0,0 +1,139 @@
+    // and the region to use with S3.
+    final S3Client wrappedClient = S3Client.builder()
+            .credentialsProvider(defaultCreds)
+            .region(Region.of(S3_REGION.toString()))
+            .build();
+
+    // Instantiate the wrapped Async S3 client with the default credentials
+    // and the region to use with S3.
+    final S3AsyncClient wrappedAsyncClient = S3AsyncClient.builder()
+            .credentialsProvider(defaultCreds)
+            .region(Region.of(S3_REGION.toString()))
+            .build();
+
+    // Instantiate the KMS client with alternate credentials.
+    // This client will be used for all KMS requests.
+    final KmsClient kmsClient = KmsClient.builder()
+            .credentialsProvider(altCreds)
+            .region(Region.of(KMS_REGION.toString()))
+            .build();
+
+    // Instantiate a KMS Keyring to use with the S3 Encryption Client.
+    final KmsKeyring kmsKeyring = KmsKeyring.builder()
+            .wrappingKeyId(ALTERNATE_KMS_KEY)
+            .kmsClient(kmsClient)
+            .build();
+
+    // Instantiate the S3 Encryption Client using the configured clients and keyring.
+    final S3Client s3Client = S3EncryptionClient.builder()
+            .wrappedClient(wrappedClient)
+            .wrappedAsyncClient(wrappedAsyncClient)
+            .keyring(kmsKeyring)
+            .build();
+
+    // Use the client to call putObject.
+    s3Client.putObject(builder -> builder
+                    .bucket(BUCKET)
+                    .key(objectKey)
+                    .build(),
+            RequestBody.fromString(input));
+
+    // Use the client to call getObject.
+    ResponseBytes<GetObjectResponse> objectResponse = s3Client.getObjectAsBytes(builder -> builder
+            .bucket(BUCKET)
+            .key(objectKey)
+            .build());
+    String output = objectResponse.asUtf8String();
+    // Check that the output matches the input.
+    assertEquals(input, output);
+
+    // Delete the object.
+    deleteObject(BUCKET, objectKey, s3Client);
+    // Close the S3 Client.
+    s3Client.close();
+  }
+
+  /**
+   * This example demonstrates how to use a single client configuration for
+   * the S3 and KMS clients used within the S3 Encryption Client.
+   */
+  public static void TopLevelClientConfiguration() {
+    final String objectKey = appendTestSuffix("top-level-client-configuration-example");
+    final String input = "TopLevelClientConfigurationExample";
+    // Load your AWS credentials from an external source.
+    final AwsCredentialsProvider creds = new S3EncryptionClientTestResources.AlternateRoleCredentialsProvider();
+
+    // Instantiate the S3 Encryption Client via its builder.
+    // By passing the creds into the credentialsProvider parameter,
+    // the S3EC will use these creds for both S3 and KMS requests.
+    final S3Client s3Client = S3EncryptionClient.builder()
+            .credentialsProvider(creds)
```

Comment:
This is more like a default credentials provider, right?
If I were to pass in my own S3/KMS clients *and* a credentials provider, the clients' creds would *not* be overridden -- correct?

Reply:
good point,
the wrapped client takes precedence. 
i'll make a note in the Javadoc comments.